src/App.js

```diff
@@ -1,75 +1,127 @@
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">sddfgdfgdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-        </ul>
-      </section>
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Form } from './components/Form/Form';
+
+class App extends React.Component {
+  state = {
+    todos: [],
+    originTodos: [],
+    indexTab: false,
+  };
+
+  addTodo = (todo) => {
+    this.setState(prevState => ({
+      originTodos: [...prevState.todos, todo],
+      todos: [...prevState.todos, todo],
+    }));
+  };
+
+  filteredActive = () => {
+    this.setState(prevState => ({
+      todos: [...prevState.originTodos].filter(elem => elem.completed === false),
+      indexTab: 'active',
+    }));
+  };
+
+  filteredCompleted = () => {
+    this.setState(prevState => ({
+      todos: [...prevState.originTodos].filter(elem => elem.completed === true),
+      indexTab: 'completed',
+    }));
+  };
+
+  handleReset = () => {
+    this.setState(prevState => ({
+      todos: [...prevState.originTodos],
+      indexTab: false,
+    }));
+  };
+
+  handleDelete = (todo) => {
+    this.setState(prevState => ({
+      todos: [...prevState.todos].filter(elem => elem.id !== todo.id),
+    }));
+  };
+
+  handleClearCompleted = () => {
+    this.setState(prevState => ({
+      todos: [...prevState.todos].filter(todo => !todo.completed),
+      originTodos: prevState.originTodos.filter(todo => !todo.completed),
+    }));
+  };
+
+  checkBoxClick = (index) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.map((elem, i) => (i === index
+        ? Object.assign(elem, { completed: !prevState.todos[i].completed })
+        : elem)),
+    }));
 
-      <footer className="footer" style={{ display: 'block' }}>
-        <span className="todo-count">
-          3 items left
-        </span>
-
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
-
-          <li>
-            <a href="#/active">Active</a>
-          </li>
-
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
-
-        <button
-          type="button"
-          className="clear-completed"
-          style={{ display: 'block' }}
-        />
-      </footer>
-    </section>
-  );
+    if (this.state.indexTab === 'active') {
+      this.filteredActive();
+    }
+
+    if (this.state.indexTab === 'completed') {
+      this.filteredCompleted();
+    }
+  };
+
+  handleAllcompleted = () => {
+    this.setState(prevState => ({
+      todos:
+        prevState.todos.every(item => item.completed)
```

Comment:
are you sure you need such a complicated structure?

Reply:
if you don't follow the comment, explain why right here

Reply:
I explained to you why I think so